source/MaterialXTest/ShaderValid.cpp

```diff
@@ -1022,8 +1060,104 @@ TEST_CASE("MaterialX documents", "[shadervalid]")
+
+
+    if (options.checkImplCount)
+    {
+        profilingLog << "---------------------------------------" << std::endl;
+
+        // Get implementation count from libraries
+        std::set<mx::ImplementationPtr> libraryImpls;
+        const std::vector<mx::ElementPtr>& children = dependLib->getChildren();
+        for (auto child : children)
+        {
+            mx::ImplementationPtr impl = child->asA<mx::Implementation>();
+            if (!impl)
+            {
+                continue;
+            }
+            libraryImpls.insert(impl);
+        }
+
+        const std::unordered_map<std::string, mx::ShaderNodeImplPtr>* oslImpls = nullptr;
+#ifdef MATERIALX_BUILD_GEN_GLSL
+        oslImpls = &(oslShaderGenerator->getImplementationsUsed());
+#endif
+        const std::unordered_map<std::string, mx::ShaderNodeImplPtr>* glslImpls = nullptr;
+#ifdef MATERIALX_BUILD_GEN_OSL
+        glslImpls = &(glslShaderGenerator->getImplementationsUsed());
+#endif
+        size_t skipCount = 0;
+        profilingLog << "-- Possibly missed implementations ----" << std::endl;
+        std::vector<std::string> whiteList =
+        {
+            "arrayappend", "backfacing", "screen", "curveadjust", "dot_surfaceshader", "mix_surfaceshader"
+            "displacementShader", "displacementshader", "volumeshader", "IM_dot_filename", "ambientocclusion", "dot_lightshader"
+        };
+        const std::string OSL_STRING("osl");
+        const std::string SX_OSL_STRING("sx_osl");
+        for (auto libraryImpl : libraryImpls)
+        {
+            const std::string& implName = libraryImpl->getName();
+
+            // Skip white-list items
+            bool whileListFound = false;
+            for (auto w : whiteList)
+            {
+                if (implName.find(w) != std::string::npos)
+                {
+                    skipCount++;
+                    whileListFound = true;
+                    break;
+                }
+            }
+            if (whileListFound)
+            {
+                continue;
+            }
+
+            if (usedImpls.count(implName))
+            {
+                continue;
+            }
+
+            if (oslImpls && oslImpls->count(implName))
+            {
+                continue;
+            }
+            if (glslImpls && glslImpls->count(implName))
+            {
+                continue;
+            }
+
+            // See if we have a sx-osl implementation used
+            // instead of the reference one
+            if (libraryImpl->getLanguage() == OSL_STRING)
+            {
+                size_t endSize = implName.size() - 3;
+                if (endSize > 0)
+                {
+                    std::string ending = implName.substr(endSize);
+                    if (ending == OSL_STRING)
+                    {
+                        std::string sxImplName = implName.substr(0, endSize) + SX_OSL_STRING;
+                        if (oslImpls->count(sxImplName))
```

Comment:
Does a basic scan by string substitution of _osl with _sx_osl which isn't fool-proof but easier than trying to find the original nodedefs (if it were possible) and rescan for corresponding implementations again.